### DISCUSSION TITLE
Thermostat AVATTO ZWT81 (ME81H)

### DISCUSSION THREAD

**Discussion by shymchenko** [Koenkk/zigbee2mqtt#10262] [Devices]
https://www.aliexpress.com/item/1005002960679997.html?spm=a2g0s.9042311.0.0.77db4c4dqgR2eG
Thermostat is not supported by Zigbee2Mqtt
I made an external integration 

https://drive.google.com/file/d/1JcqlLF61wOlUBCgggvUvYdV--SCWS0x_/view?usp=drivesdk

The main functions are all working!
But I can not set and get deadzone_temperature и max_temperature_limit!
State:
{
    "child_lock": "UNLOCK",
    "current_heating_setpoint": 15,
    "heat": "OFF",
    "linkquality": 68,
    "local_temperature": 21,
    "local_temperature_calibration": 0,
    "outdoor_temperature": 20,
    "preset": "program",
    "program": [
        {
            "weekdays_p1": "6h:0m 20°C"
        },
        {
            "weekdays_p2": "8h:0m 15°C"
        },
        {
            "weekdays_p3": "11h:30m 15°C"
        },
        {
            "weekdays_p4": "12h:30m 15°C"
        },
        {
            "weekdays_p5": "17h:0m 22°C"
        },
        {
            "weekdays_p6": "22h:0m 15°C"
        },
        {
            "saturday_p1": "8h:0m 22°C"
        },
        {
            "saturday_p2": "23h:0m 15°C"
        },
        {
            "sunday_p1": "8h:0m 22°C"
        },
        {
            "sunday_p2": "23h:0m 15°C"
        }
    ],
    "running_state": "idle",
    "sensor": "AL",
    "state": "ON",
    "system_mode": "heat"
}

No events with  data points tuya.dataPoints.moesMaxTempLimit or tuya.dataPoints.moesMaxTemp or tuya.dataPoints.moesDeadZoneTemp

Could you please help me to solve that problem

**Comment by malcharek** [comment]
Hi shymchenko,

Please excuse me for simply replying here. I am completely new to the subject of Zigbee and also got myself a ME81H. Zigbee2mqtt and a MQTT broker I already have running. My only problem is that I do not know how to connect the ME81H. Do you have any special gateway or something? How did you get the initial connection working? In my logfiles I see that zigbee2mqtt is running but I don't see any requests from other devices.

I would be really happy to read from you. Thanks
Michael

**Reply by shymchenko** [reply]
> Hi shymchenko,
> 
> Please excuse me for simply replying here. I am completely new to the subject of Zigbee and also got myself a ME81H. Zigbee2mqtt and a MQTT broker I already have running. My only problem is that I do not know how to connect the ME81H. Do you have any special gateway or something? How did you get the initial connection working? In my logfiles I see that zigbee2mqtt is running but I don't see any requests from other devices.
> 
> I would be really happy to read from you. Thanks Michael

https://www.zigbee2mqtt.io/guide/faq/

**Comment by Westcott1** [comment]
Try this converter -
[https://neatho.me/ru/help/termostat-avatto-zwt81/](url)

**Comment by fulup-bzh** [comment]
@shymchenko the google drive you linked is not valid any more. I have the same device and would be interested by your integration. Could you please post your code/config on git. Same thing go neatho.me link that is only not valid anymore.
Thankyou

**Comment by shymchenko** [comment]
https://drive.google.com/file/d/1JcqlLF61wOlUBCgggvUvYdV--SCWS0x_/view?usp=drivesdk

**Comment by fulup-bzh** [comment]
Thank you very much, I hopefully will get time during Xmast vacaction to check this.

**Comment by Cib0rg** [comment]
Currently z2m throw error with this converter
`zigbee2mqtt-1  | [2025-06-28 23:39:33] error:   z2m: Invalid external converter 'avatto_me81h.mjs' was ignored and renamed to prevent interference with Zigbee2MQTT.`
Looks like there was some major changes in past 3 years. What should I do to make it work?
